bgpd/bgp_network.c

```diff
@@ -438,12 +438,15 @@ static int bgp_accept(struct thread *thread)
 		return -1;
 	}
 
-	/* Check whether max prefix restart timer is set for the peer */
-	if (peer1->t_pmax_restart) {
+	/* Do not try to reconnect if the peer reached maximum
+	 * prefixes, restart timer is still running or the peer
+	 * is shutdown.
+	 */
+	if (BGP_PEER_START_SUPPRESSED(peer1)) {
```

Comment:
with BGP_PEER_START_SUPPRESSED, it will check PEER_FLAG_SHUTDOWN & PEER_STATUS_PREFIX_OVERFLOW flags. But what if peer restart timer is still running ? Now code doesn't check this any more. Isn't this an issue ?

Btw, Apart from this place, this timer is not used any where else as well.

Reply:
I thought about using:
```
if (peer1->t_pmax_restart || (BGP_PEER_START_SUPPRESSED(peer1) && !peer1->t_pmax_restart)) {
   ...
}
```
do you think it would be the right way?

Reply:
Otherwise, PEER_STATUS_PREFIX_OVERFLOW is set while t_pmax_restart is running. It's cleared after the timer expires thus it's enough to have BGP_PEER_START_SUPPRESSED here.